lib/services.js

```diff
@@ -218,6 +218,19 @@ export default {
             authInfo.getAccountDisplayName();
         // This should be object creator's canonical ID.
         omVal['owner-id'] = authInfo.getCanonicalID();
+        // TODO: Find a way to store these headers for multipart uploads.
+        // params.headers is undefined when called from
+        // completeMultipartUploads because the latter does not pass
+        // headers as part of the params object
```

Comment:
We will probably decide to not use the headers to store this but rather send in the values directly as params.  For now, however, this is good to add the tests and get us compliant and I will help you with the follow up that will include MPU.  Note that for content-encoding we need to modify the value to remove 'aws-chunked' if there.  In streaming v4 auth, this value is sent in but AWS says they do not store it along with any other content encoding value.  So, please check whether there is 'aws-chunked' in the value and remove it.  See http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html and search for "content-encoding".